_episodes/05-scripts.md

```diff
@@ -83,41 +83,44 @@ $ ls -l
-      `yourUsername`).
-    + The second set of `rwx`s are permissions that other members of the owner's group share (in
-      this case, the group is named `tc001`).
-    + The third set of `rwx`s are permissions that anyone else with access to this computer can do
-      with a file. Though files are typically created with read permissions for everyone, typically
-      the permissions on your home directory prevent others from being able to access the file in
-      the first place.
-+ **2nd column - Owner:** This is the username of the user who owns the file. Their permissions are
-  indicated in the first permissions column.
-+ **3rd column - Group:** This is the user group of the user who owns the file. Members of this user
-  group have permissions indicated in the second permissions column.
-+ **4th column - Size of file:** This is the size of a file in bytes, or the number of
-  files/subdirectories if we are looking at a directory. (We can use the `-h` option here to get a
-  human-readable file size in megabytes, gigabytes, etc.)
-+ **5th column - Time last modified:** This is the last time the file was modified.
-+ **6th column - Filename:** This is the filename.
+That's a huge amount of output: a full listing of everything in the directory. Let's see if we can
+understand what each field represents, working left to right.
+
+1. **Permissions:** On the very left side, there is a string of the characters `d`, `r`, `w`, `x`,
+   and `-`. The `d` indicates if something is a directory (there is a `-` in that spot if it is not
+   a directory). The other `r`, `w`, `x` bits indicates permission to **R**ead, **W**rite, and
+   e**X**ecute a file. There are three fields of `rwx` permissions following the spot for `d`. If a
+   user is missing a permission to do something, it's indicated by a `-`.
+     + The first set of `rwx` are the permissions that the owner has (in this case the owner is
+       `yourUsername`).
+     + The second set of `rwx`s are permissions that other members of the owner's group share (in
+       this case, the group is named `tc001`).
+     + The third set of `rwx`s are permissions that anyone else with access to this computer can do
+       with a file. Though files are typically created with read permissions for everyone, typically
+       the permissions on your home directory prevent others from being able to access the file in
+       the first place.
+2. **References:** This counts the number of references (hard links) to the item (file, folder,
+   symbolic link or "shortcut").
+3. **Owner:** This is the username of the user who owns the file. Their permissions are indicated
+   in the first permissions field.
+4. **Group:** This is the user group of the user who owns the file. Members of this user group have
+   permissions indicated in the second permissions field.
+5. **Size of file:** This is the size of a file in bytes, or the number of files/subdirectories if
```

Comment:
The claim with reference to folders irritates me. Here is an example of a synthetic folder:
```
$ tree .       
.
├── bar
├── baz.log
└── foo
    ├── baz
    └── baz.txt
$ \ls -l
total 0
drwxr-xr-x. 2 username username 40 Jul 16 10:03 bar
-rw-r--r--. 1 username username  0 Jul 16 10:03 baz.log
drwxr-xr-x. 3 username username 80 Jul 16 10:03 foo
```
the field between `username` and `Jul 16` doesn't provide the number of files/subdirs if I am not mistaken. 
I checked the manpage of `ls` and was stunned to find **NO** hint to what `ls -l` actually prints out. :-1: 

A quote from `man ls` on my system (fc31, kernel `5.7.7-100.fc31.x86_64`):
```
       -l     use a long listing format
```

Reply:
[IBM](https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.1.0/com.ibm.zos.v2r1.bpxa500/lscmd.htm#lscmd__longf) indicates it's the *number of filesystem blocks* occupied by the contents of the directory.

Reply:
Ok, should we add that here or is that too much mental overhead? Balancing completeness versus simplicity.

Reply:
Yes, please add that here :-)